skbeam/core/accumulators/arrayProjector.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+
+import math
+import numpy as np
+
+"""
+authors: Mikhail Dubrovin, TJ Lane, Christopher O'Grady
+"""
+
+class ArrayProjector(object):
+
+    def __init__(self, bin_values, weights=None, nbins=101, norm=True):
+        """
+        Parameters
+        ----------
+        bin_values : np.ndarray
+            For each pixel, this is the value of that determines which bin
+            a pixel's intensity will contribute to
+        weights : np.ndarray
+            A weight for each pixel with the same shape as bin_values (can be zero to ignore a pixel)
+        nbins : int
+            The number of bins to employ. If `None` guesses a good value.
```

Comment:
docstring and code are inconsistent. There is no `if nbins is None:` handling in the `__init__`.  Given the current code, this should probably just read `The number of bins to employ`.  I am 50/50 on adding heuristics to guess a good value of nbins. numpy recently picked up an `auto` keyword for its histogram function.  You may want to consider using their machinery for guessing an appropriate number of bins. https://github.com/numpy/numpy/blob/master/numpy/lib/function_base.py#L300